plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/catalog/glue/GlueIcebergTableOperations.java

```diff
@@ -102,13 +104,36 @@ protected String getRefreshedLocation(boolean invalidateCaches)
     protected void commitNewTable(TableMetadata metadata)
     {
         verify(version.isEmpty(), "commitNewTable called on a table which already exists");
+        CommitStatus status = CommitStatus.FAILURE;
         String newMetadataLocation = writeNewMetadata(metadata, 0);
         TableInput tableInput = getTableInput(tableName, owner, ImmutableMap.of(METADATA_LOCATION_PROP, newMetadataLocation));
 
         CreateTableRequest createTableRequest = new CreateTableRequest()
                 .withDatabaseName(database)
                 .withTableInput(tableInput);
-        stats.getCreateTable().call(() -> glueClient.createTable(createTableRequest));
+        try {
+            stats.getCreateTable().call(() -> glueClient.createTable(createTableRequest));
+            status = CommitStatus.SUCCESS;
+        }
+        catch (AmazonServiceException e) {
+            if (e instanceof OperationTimeoutException) {
+                status = CommitStatus.UNKNOWN;
+            }
+            throw e;
+        }
+        catch (RuntimeException e) {
+            // Check whether the creation of the table eventually succeeded
+            status = checkCommitStatus(newMetadataLocation);
+            switch (status) {
+                case FAILURE -> throw e;
```

Comment:
This line is unreachable, right?
`checkCommitStatus` never returns FAILURE.

rename `checkCommitStatus` to `checkCommitVisibilySuccesful` and make it return boolean
(the "Visibily" is because the commit might be succesful but the method may still return false, if there were a few more commits in the meantime, and our own metadata location got forgotten)

Reply:
BTW i think the additional call to check commit status is redundant. In exception handling path it's better not to do it, as we may obliterate original failure.
I think it should be sufficient to inspect AmazonServiceException for known obvious failures and let the metadata file be deleted when they happen, and leave it behind otherwise. Leaving it behind is not a very bad thing (it will get cleaned up eventually), so we don't need to try very very hard to prevent it from happening.